### PR TITLE
Backoffice Order Address Copying

### PR DESCRIPTION
**Pull request by mwrouse**
Added a button in the back office for copying the entire address separated by `\n`

Double clicking each line highlights a word (browser dependent) and it will copy only what is highlighted. Triple clicking highlights the entire line (browser dependent) and again will only copy what is highlighted.

This allows you to quickly copy the entire line or just part of the line (example, just the postal code).

image

There are also now some helpful functions for copying stuff to the clipboard.

For Issue #1413

### CODE REVIEW

File: admin-dev/themes/default/template/controllers/orders/helpers/view/view.tpl [hunk 1]
```diff
@@ -709,11 +709,19 @@
...
+                        {l s='Copy'}
+                      </a>
                       <a class="btn btn-default pull-right" href="?tab=AdminAddresses&amp;id_address={$addresses.delivery->id}&amp;addaddress&amp;realedit=1&amp;id_order={$order->id}&amp;address_type=1&amp;token={getAdminToken tab='AdminAddresses'}&amp;back={$smarty.server.REQUEST_URI|urlencode}">
                         <i class="icon-pencil"></i>
                         {l s='Edit'}
                       </a>
-                      {displayAddressDetail address=$addresses.delivery newLine='<br />'}
+
+                      <div ondblclick="copyDivTextContentToClipboard(event, true);" data-placement="left" data-content="{l s='Copied!'}" data-trigger="manual">
+                      {displayAddressDetail address=$addresses.delivery newLine='</div><div ondblclick="copyDivTextContentToClipboard(event, true);" data-placement="left" data-content="Copied!" data-trigger="manual">'}
```
Comment (mwrouse) [inline admin-dev/themes/default/template/controllers/orders/helpers/view/view.tpl:722]: How does one use the `l` function inside of something like this? Should definitely pull variations of `Copied!` from the language. I'm just not sure how.

**Comment by mwrouse** [comment]
I agree that it is not ideal for everything, obviously there will always be differences in wherever you are pasting the address. 

The solution implemented here would have been for copying each line directly into something.

Maybe the solution would be to:
1. Auto-copy whatever gets highlighted when you double click
2. Auto-copy the entire line when you triple click

**Comment by therampagerado** [comment]
Is there such option?

**Comment by mwrouse** [comment]
Yes, absolutely. I just pushed changes, you'll need to change stuff in both `tools.js` and `view.tpl` again.

**Comment by therampagerado** [comment]
This works just perfect now!

**Comment by getdatakick** [comment]
Thanks guys. I have picked this via commit 05affeeeb438431cc5582d671785b9ce3b9ca0a3.

**Comment by getdatakick** [comment]
Small modifications were needed

1) fallback to `document.execCommand('copy');` implementation for older browser, or for non-secured context (for example back office served via http protocol)
2) address escaping - since address is rendered inside `onclick="...textaddr..."`, we need to make sure `textaddr` does not contains any html characters, especially `"`

Otherwise, very neat.

**Pull request closed without merging**